api/models/author.py

```diff
@@ -48,44 +49,82 @@ def _get_cached_author(self, instance):
             # TODO: Should be an exception
             return None
 
-    def follow(self, author):
+    def _get_author(self, instance):
+        if isinstance(instance, Author):
+            return instance
+        if isinstance(instance, CachedAuthor):
+            return Author.objects.get(id=instance.id)
+        else:
+            return None
+
+    def add_friend(self, friend):
+        # Check if the potential friend has already befriended the author
+        friend = self._get_cached_author(friend)
```

Comment:
Because in order for this to support remote friends. We need to short circuit and assume success rather than blowing up the API call.